### PR TITLE
Permagunk Revival

### PR DESCRIPTION
**Pull request by bathosbathtime**
Unironically copy/pasted from #21632 
Please review with care.

This PR adds permagunk, an idea from @I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE and reopened by myself.  I will do my best to fix any conflicts and add any changes requested by the review process.  Trash and cleanable messes now persist for up to five rounds.  Admins can disable this from a new verb.  Excluded from central command, shuttles, and vaults.  Does not include fuel, soot, or polyacid goo.  This also should add more cleaning grenades to janiclosets, one for every 100 persistent gunk tiles per janitor closet.
Compiled and tested lovingly locally.
Full credit to I-VAP-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE and also MistakeNot4892
:cl:
 * rscadd: Permagunk is now enabled.  Garbage and cleanable messes will now last up to five rounds.
 * rscadd: Any round ending antag that destroys the station, like nuke ops or cult, will cause the next round on that map to be shiny and clean.
 * rscadd: Janitors get one extra cleaning grenade in each janitor closet (room) per 100 gunk tiles from the previous round

### CODE REVIEW

**Comment by Intigracy** [comment]
I'll be merging this tomorrow, Sunday 10MAR.

**Comment by Kurfursten** [comment]
You better review it first

**Comment by spoonylove** [comment]
Do dead bees persist?

**Comment by Intigracy** [comment]
Reviewing it is why I'm going to merge it tomorrow instead of today.

Have faith, ye of little faith.

**Comment by bathosbathtime** [comment]
That is a good question.  They do not.

**Comment by spoonylove** [comment]
That is unfortunate.

**Comment by Coxswain-Navigator** [comment]
This can't possibly go wrong, especially considering turbogunked rounds and how much lag filth generates. Not to mention, cult.

**Comment by bathosbathtime** [comment]
Before merging, I might need to change the changelog.  This will still be a test merge, correct?

**Comment by DeityLink** [comment]
@Cockswain it was made so that cult victories and a few other cause gunk to not persist.

**Comment by Intigracy** [comment]
Blood generates lag only because of all of the associated vars with it.

After a round ends and it's still there, it loses all of the associated dna vars and shit.

**Comment by Intigracy** [comment]
Yeah, anything that destroys the station doesn't persist any gunk. Clean station.

I'm not seeing anything on here that should break anything, but it'll probably break something.

**Review by Intigracy — APPROVED** [review]

**Comment by Intigracy** [comment]
Finally the ~~rats~~ assistants spawn in an ~~trashheap~~ area fitting of their stature.

**Comment by SonixApache** [comment]
Should totally use the persistence system to show the previous round's roundend in the next game's lobby when the [SHOW ROUNDEND] button is pressed.

**Comment by DeityLink** [comment]
Yes please.

File: code/game/objects/effects/decals/cleanable.dm [hunk 1]
```diff
@@ -17,11 +17,77 @@
...
+		src.pixel_x = pixel_x
+	if(pixel_y)
+		src.pixel_y = pixel_y
+
+	if(ticker)
+		initialize()
+
+	..(loc)
+
+	blood_list += src
```
Comment (Zth--) [inline code/game/objects/effects/decals/cleanable.dm:45]: man... you need to clean up this stuff on Destroy()

Comment (Zth--) [reply code/game/objects/effects/decals/cleanable.dm:45]: nevermind, there's two `/obj/effect/decal/cleanable/Destroy()`
fuck me

Comment (Kurfursten) [reply code/game/objects/effects/decals/cleanable.dm:45]: I have said it before and I'll say it again: multiple declarations of a proc in separate files are a mistake which should be purged

**Pull request merged**